### PR TITLE
Add instrumentation for cats effect

### PR DESCRIPTION
**Pull request by jatcwang**
Aspectj instrumentation is working (test passing) but not with Kanela. I think I'm doing everything right by following the pattern set out in the other modules so I may need some help for this!

### CODE REVIEW

File: kamon-cats-io/src/main/scala/kamon/instrumentation/catseffect/CatsIoInstrumentation.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import kamon.Kamon
+import kamon.context.{Context, Storage}
+import kamon.instrumentation.Mixin.HasContext
+import kanela.agent.api.instrumentation.mixin.Initializer
+import kanela.agent.libs.net.bytebuddy.asm.Advice
+import kanela.agent.scala.KanelaInstrumentation
+
+class CatsIoInstrumentation extends KanelaInstrumentation {
+
+  forTargetType("cats.effect.internals.IOShift.Tick") { builder =>
```
Comment (dpsoft) [inline kamon-cats-io/src/main/scala/kamon/instrumentation/catseffect/CatsIoInstrumentation.scala]: hey @jatcwang awesome!!. The only thing you need for the instrumentation to work is:

change this: 
```scala 
  forTargetType("cats.effect.internals.IOShift.Tick") 
```
to 

```scala
forTargetType("cats.effect.internals.IOShift$Tick")
```

Comment (jatcwang) [reply kamon-cats-io/src/main/scala/kamon/instrumentation/catseffect/CatsIoInstrumentation.scala]: Ah haha I tried IOShift$.Tick :pray:

**Comment by jatcwang** [comment]
@dpsoft Thanks for the suggestion. Instrumentation now working with Kanela agent too. CI is currently failing though because cats-effect isn't published for Scala 2.10..

**Comment by last-ent** [comment]
Hi, is this PR going to be merged? It would be really awesome to have this feature in Kamon

**Comment by ivantopo** [comment]
Hey @jatcwang, thanks a lot for bring this in! I'll publish a release shortly :tada: 

/cc: @last-ent @dpsoft

**Comment by ivantopo** [comment]
Published as version `1.1.0` at https://bintray.com/kamon-io/releases/kamon-futures and now on its way to Maven Central!

**Pull request merged**